### PR TITLE
[APB-1905][PT, JR]: Implemented VrnValidation and added tests.

### PR DESCRIPTION
**Pull request by JTY009**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/agentmtdidentifiers/model/VrnValidation.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  private[model]def weightedTotal(reference: String): Int = {
+    val weighting = List(8, 7, 6, 5, 4, 3, 2)
+    val ref = reference.map(_.asDigit).take(7)
+    (ref, weighting).zipped.map(_ * _).sum
+  }
+
+  private[model] def takeCheckSumPart(vrn: String): Int = vrn.takeRight(2).toInt
+
+  def isValid(vrn: String): Boolean = {
+    val total = weightedTotal(vrn)
```
Comment (opetch) [inline src/main/scala/uk/gov/hmrc/agentmtdidentifiers/model/VrnValidation.scala]: We should check that the string is 9 characters long and all digits

Comment (JTY009) [reply src/main/scala/uk/gov/hmrc/agentmtdidentifiers/model/VrnValidation.scala]: True, will update this

File: src/main/scala/uk/gov/hmrc/agentmtdidentifiers/model/VrnValidation.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    if (takeCheckSumPart(vrn) == calcCheckSum97(total)) true
```
Comment (opetch) [inline src/main/scala/uk/gov/hmrc/agentmtdidentifiers/model/VrnValidation.scala]: if you wish these 3 lines can be condensed as so...
```if (takeCheckSumPart(vrn) == calcCheckSum97(total)) true else takeCheckSumPart(vrn) == calcCheckSum9755(total)```

Comment (JTY009) [reply src/main/scala/uk/gov/hmrc/agentmtdidentifiers/model/VrnValidation.scala]: Have updated, hope it is what you asked

**Pull request merged**